tools/reckless

```diff
@@ -588,16 +592,115 @@ class InferInstall():
         raise Exception(f'plugin entrypoint not found in {self.dir}')
 
 
-python3pip = Installer('python3pip', 'text/x-python', exe='python3',
-                       manager='pip', entry='{name}.py')
-python3pip.add_entrypoint('{name}')
-python3pip.add_entrypoint('__init__.py')
-python3pip.add_dependency_file('requirements.txt')
-python3pip.add_dependency_call(['pip', 'install', '-r', 'requirements.txt'])
+class InstallationFailure(Exception):
+    "raised when pip fails to complete dependency installation"
 
-python3pip3 = python3pip.copy()
-python3pip3.manager = 'pip3'
-python3pip3.dependency_call = [['pip3', 'install', '-r', 'requirements.txt']]
+
+def create_python3_venv(staged_plugin: InstInfo) -> InstInfo:
+    "Create a virtual environment, install dependencies and test plugin."
+    logging.debug("configuring a python virtual environment")
+    env_path = Path('.venv')
+    env_path_full = Path(staged_plugin.source_loc) / env_path
+    plugin_path = Path(staged_plugin.source_loc) / 'source'
+
+    # subdir should always be None at this point
+    if staged_plugin.subdir:
+        logging.debug("warning: cloned plugin contains subdirectory")
+        plugin_path = plugin_path / staged_plugin.subdir
```

Comment:
How this can work with core lightning?

If the plugin installs the dependencies inside the .venv, core lightning should run in the same one right?

I think the only approach that can work, is what it is used in coffee https://github.com/coffee-tools/coffee/blob/master/coffee_lib/src/plugin.rs#L52-L65

Reply:
Each plugin gets its own virtual environment.  There's a wrapper that initializes the environment for that plugin from the shebang.  Core lightning and the plugin are independent processes, so lightningd doesn't care if it's running in a different environment.  I think it's more resilient to use isolated environments in order to avoid conflicting dependencies.  Also, now we don't have to worry about which environment Core Lightning was using, because reckless has no concept of this.

Reply:
>There's a wrapper that initializes the environment for that plugin from the shebang.

So I think this deserve to be written somewhere to understand that reckless do not run the main file directly. It is better also for us to know that to easily help people that will have problem with plugins in the future